modules/apps/analytics/analytics-settings-web/src/main/java/com/liferay/analytics/settings/web/internal/portlet/action/AddChannelMVCRenderCommand.java

```diff
@@ -39,12 +39,14 @@ protected String getDefaultJspPath() {
 		return "/add_channel.jsp";
 	}
 
+	@Override
+	protected ServletContext getServletContext() {
+		return _servletContext;
+	}
+
 	@Reference(
-		target = "(osgi.web.symbolicname=com.liferay.analytics.settings.web)",
-		unbind = "-"
+		target = "(osgi.web.symbolicname=com.liferay.analytics.settings.web)"
```

Comment:
@JanisZhang It seems all these classes are referencing the same servlet context? For this, I think you can keep the `protected volatile ServletContext` field in the base class, and directly add the `@Reference` annotation to that field (this module has `-dsannotations-options: inherit` by default). So in this way we can just remove all these set methods.